src/app/tabs/runTab/contractDropdown.js

```diff
@@ -43,16 +48,69 @@ class ContractDropdownUI {
     })
   }
 
+  listenToContextChange () {
+    this.blockchain.event.register('newExecutionContext', () => {
+      this.blockchain.updateNetwork((err, {name} = {}) => {
+        if (err) {
+          console.log(`can't detect network`)
+          return
+        }
+        this.exEnvironment = this.blockchain.getProvider()
+        this.networkName = name
+
+        const savedConfig = window.localStorage.getItem(`ipfs/${this.exEnvironment}/${this.networkName}`)
+
+        // check if an already selected option exist else use default workflow
+        if (savedConfig !== null) {
+          this.setCheckedState(savedConfig)
+        } else {
+          if (this.exEnvironment === 'vm') {
+            this.setCheckedState(false)
+          } else if (this.exEnvironment === 'web3') {
```

Comment:
even if `web3` is selected, could be that the network your local node is syncing is `Main`.
`this.setCheckedState(this.networkName === 'Main')` should be fine